src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,25 @@
-import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { FilterOption } from '../../enums/FilterOptions';
+import { actions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const query = useAppSelector(state => state.filter.query);
+  const dispatch = useDispatch();
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            onChange={(e) => {
+              dispatch(actions.setStatus(e.target.value as FilterOption));
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function).